src/averageAges.js

```diff
@@ -55,9 +64,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const motherAndSon = people.filter(({ mother, sex }) =>
+    people.some(mom => mom.name === mother && (
+      !onlyWithSon || sex === 'm'
+    )));
+
+  const averageAgeDiff = motherAndSon.map(({ born, mother }) => {
+    const kidMom = people.find(mom => mom.name === mother);
+
+    return born - kidMom.born;
+  });
+
+  return averageValue(averageAgeDiff);
 }
 
+const averageValue = (numbers) => {
```

Comment:
Good Job! By checklist name of the function has to start from the verb)

Reply:
thanx!